core/snmp/proxy-rpc-impl/src/main/java/org/opennms/netmgt/snmp/proxy/common/SnmpProxyRpcModule.java

```diff
@@ -176,7 +176,8 @@ private static final void addResult(SnmpResult result, String correlationId, Map
         final CompletableFuture<SnmpValue[]> future = SnmpUtils.getAsync(request.getAgent(), oids);
         return future.thenApply(values -> {
             final List<SnmpResult> results = new ArrayList<>(oids.length);
-            for (int i = 0; i < oids.length; i++) {
+            int len = Math.min(oids.length, values.length);
```

Comment:
if `oids.length != values.length` then the oid/value mapping bellow will not be accurate

Reply:
True, was discussing with @agalue. He proposed to compare OIDs coming from the result and map accordingly.  That seems better approach

Reply:
scratch that, since we don't carry request PDU in SNMPValue, it may be good to throw exception in `SNMP4Strategy` when request and response size are not matching.

Reply:
Since returning wrong response is worse than not having response, I reverted that.
Now logging whenever response size is not matching with request size and throwing custom exception.